### PR TITLE
Resolves #92, Identifier Name Violation Issues after updating to Xcode 9.3

### LINKED ISSUE #92: Identifier Name Violation Issues and Warning after updating to Xcode 9.3
**Checklist**
- [x] Bug Report - The issue you are reporting might be there already. Please search on the [issues tracker.](https://github.com/openfoodfacts/openfoodfacts-ios/issues)
- [x] Before reporting, please ensure (if you can) that you have tested it with the latest (dev) version of the software.

**Summary:** 

Identifier Name Violation Issues and Warning after updating to Xcode 9.3

**Steps to reproduce:** 

Please provide detailed steps for reproducing the issue.

 - Update to Xcode 9.3
 -  
 -  

**Expected behavior:** 

No Errors
**Observed behavior:** 

18 Warnings and 9 Errors
**Device and iOS version:** 

Apple iPhone [7] running iOS [11.3]

**Screenshots:** 

screen shot 2018-04-27 at 10 52 12 pm

### PR DESCRIPTION
**Pull request by antsangrigoli**
## PR Description

This is in response to issue #92. When building the project in Xcode 9.3, I was getting 9 Identifier Name Violations errors. I've fixed these errors by adding // swiftlint:disable:next identifier_name as you can see in the screenshots below. Project can now be built and ran in Xcode 9.3.

Type of Changes 

- [x] Fixes Issue #92
 
### Before 
screen shot 2018-04-28 at 4 40 48 pm
screen shot 2018-04-28 at 4 40 59 pm

### After
screen shot 2018-05-04 at 3 07 19 pm
screen shot 2018-05-04 at 3 06 44 pm

 
## Checklist
 
Make sure you've done all the following (_Put an `x` in the boxes that apply._)
 
 - [x] If you have multiple commits please combine them into one commit by [squashing](https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit) them. 
 - [x] Code is well documented
 - [x] Included unit tests for new functionality
 - [x] All user-visible strings are made translatable
 - [x] Code passes Travis builds in your branch

### CODE REVIEW

**Comment by snuff4** [comment]
I had two solution to these - 

1. Either add swiftlint disable rule, or
2. Change the name of the variable.

Like, there are many warnings due to 'vc' variable, referring to ViewControllers. We can rename them to SummaryFormTableVC or NutritionTableFormVC etc

What do you propose @antsangrigoli @roger-tan ?

File: Sources/Extensions/UIColor.swift [hunk 1]
```diff
@@ -19,10 +19,12 @@ extension UIColor {
 
         var rgb: UInt32 = 0
 
+        // swiftlint:disable identifier_name
         var r: CGFloat = 0.0
```
Comment (roger-tan) [inline Sources/Extensions/UIColor.swift]: Personally, I'm not a huge fan of abbreviation. Instead to disable the error. We should just rename r as red, green as green, b as blue and a as alpha.

**Comment by aleene** [comment]
I concur. I prefer large variable names as well, so it is clear what a variable represents.

File: Sources/ViewControllers/User/UserViewController.swift [hunk 2]
```diff
@@ -31,28 +31,28 @@ class UserViewController: UIViewController, DataManagerClient {
                 showProductsPendingUpload()
             }
 
-            vc = childNavigationController
+            UIVC = childNavigationController
```
Comment (snuff4) [inline Sources/ViewControllers/User/UserViewController.swift]: childNavigationVC

**Comment by snuff4** [comment]
@antsangrigoli can you ensure that everything is working after renaming?

**Comment by antsangrigoli** [comment]
@snuff4 yes everything seems to work fine. 
However, I'm getting warnings on the variable 'id'. You think I should add the swiftlint disable rule or rename 'id' ?

**Comment by snuff4** [comment]
@antsangrigoli I saw that too. [There's a issue related to this, too.](https://github.com/realm/SwiftLint/issues/231)

**Comment by antsangrigoli** [comment]
@snuff4 Just read up on it. I could add the following parameters below to the .swiftlint.yml file and this will solve the issue.

identifier_name:
  excluded:
    - id

**Comment by snuff4** [comment]
poke @roger-tan For this I think we can exclude for this. We need to finish with this PR asap, because without it, it's not working on 9.3

File: Sources/Extensions/UIViewController.swift [hunk 3]
```diff
@@ -29,8 +29,8 @@ extension UIViewController {
     }
 
     func openUrlInApp(_ url: URL, showAlert: Bool? = nil) {
-        let vc = SFSafariViewController(url: url, entersReaderIfAvailable: false)
-        present(vc, animated: true)
+        let SFSafariVC = SFSafariViewController(url: url, entersReaderIfAvailable: false)
```
Comment (roger-tan) [inline Sources/Extensions/UIViewController.swift]: Personally for local context. I would just call controller but a variable name cannot with an uppercase letter. We should keep the camel case for a name

**Comment by roger-tan** [comment]
For id. It's ok for me.

File: Sources/Extensions/UIViewController.swift (hunk 3 above)
Comment (antsangrigoli) [reply Sources/Extensions/UIViewController.swift]: So I should rename the variable SFSafariVC to sfSafariVC ?

Comment (roger-tan) [reply Sources/Extensions/UIViewController.swift]: To keep consistence inside the projet. Just sfSafariVC.

**Comment by roger-tan** [comment]
@antsangrigoli Thank you for your contributions into OpenFoodFacts. Your Pull Request is now merged .

**Pull request merged**